src/pipeline.h

```diff
@@ -47,8 +47,8 @@ struct Composite {
     input(nullptr),
     mode(VIPS_BLEND_MODE_OVER),
     gravity(0),
-    left(-1),
-    top(-1),
+    left(INT_MIN),
```

Comment:
`INT_MIN` is -32767, which could still be a valid offset. Perhaps add a `composite->hasOffset` boolean instead? This could be set in JavaScript, removing the need for the `IsUndefined` check in C++.

Reply:
Thanks for the suggestion - just implemented it!